src/HID-APIs/D3MouseAPI.hpp

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+D3MouseAPI::D3MouseAPI(void) : _button0(false)
+{
+	// Empty
+}
+
+void D3MouseAPI::begin(void)
+{
+	end();
+}
+
+void D3MouseAPI::end(void)
+{
+	_button0 = false;
+	_button1 = false;
+	_button2 = false;
+	rotate("x", 0);
+	rotate("y", 0);
+	rotate("z", 0);
+	move("x", 0);
+	move("y", 0);
+	move("z", 0);
+}
+
+void D3MouseAPI::click(int button)
+{
+	switch (button)
+	{
+	case 0:
+		_button0 = true;
+		update();
+		_button0 = false;
+		update();
+		break;
+	case 1:
+		_button1 = true;
+		update();
+		_button1 = false;
+		update();
+		break;
+	case 2:
+		_button2 = true;
+		update();
+		_button2 = false;
+		update();
+		break;
+	}
+}
+
+void D3MouseAPI::rotate(String axis, int16_t rotation)
+{
+	if (axis == "x")
+		_xRot = rotation;
+	if (axis == "y")
+		_yRot = rotation;
+	if (axis == "y")
+		_zRot = rotation;
+}
+
+void D3MouseAPI::move(String axis, int16_t pos)
+{
+	if (axis == "x")
+		_x = pos;
+	if (axis == "y")
+		_y = pos;
+	if (axis == "y")
```

Comment:
This appears to be a typo and should be `z`
```suggestion
	if (axis == "z")
```